lib/main/atom/views/outline/semanticViewPane.ts

```diff
@@ -0,0 +1,74 @@
+import {CompositeDisposable} from "atom"
+import {SemanticView} from "./semanticView"
+import {Disposable} from "atom"
+
+class SemanticViewPane {
+  subscriptions: CompositeDisposable
+
+  constructor(public view?: SemanticView) {
+    this.subscriptions = new CompositeDisposable()
+
+    this.subscriptions.add(
+      new Disposable(() => {
+        if (this.view) {
+          atom.workspace.hide(this.view)
+          this.view.destroy()
+        }
+      }),
+      atom.config.observe("atom-typescript.showSemanticView", val => {
+        if (val) this.show()
+        else this.hide()
+      }),
+    )
+  }
+
+  destroy() {
+    this.subscriptions.dispose()
+  }
+
+  async toggle(): Promise<void> {
+    if (!this.view) await this.show()
+    else await atom.workspace.toggle(this.view)
+  }
+
+  async show(): Promise<void> {
+    if (!this.view) this.view = new SemanticView({})
```

Comment:
:warning: Since `SemanticView` could've been deserialized, need to check for that with `atom.workspace.itemForURI`. Otherwise a real chance we'll be creating a second one.

Reply:
hmm, I will add a check in `show()`, if an instance is already present in one of the Panes

but still, I don't think the _leak_ is of big concern:  
first, it wouldn't happen often, and second, even though the `SemanticViewPane` / `SemanticViewController` will be bound in an additional `Disposable`, this `Disposable` should get garbage-collected without problems, since it is not referenced anywhere (cf. deserialization function `deserializeSemanticView` in `semanticView.ts`)

If it is still a concern for you, I could take additional measures to avoid creation of the additional `Disposable` -- should I?

Reply:
I guess this boils down to using the singleton pattern properly...

So here's an idea, implement the singleton pattern directly: make `SemanticView` constructor private, add a `create` static function and `instance` static private property. Then there would be no need to concern ourselves with whether `SemanticView` was created or not, since it wouldn't be possible to create a second instance *at all*.

Note though that it should be properly destroyed, i.e. `instance` should be set to `null`/`undefined` in destructor.